pkg/miscellaneous/miscellaneous.go

```diff
@@ -442,17 +442,22 @@ func DoPromoteMember(ctx context.Context, member *etcdserverpb.Member, cli etcdC
 	memPromoteCtx, cancel := context.WithTimeout(ctx, brtypes.DefaultEtcdConnectionTimeout)
 	defer cancel()
 
+	start := time.Now()
 	//Member promote call will succeed only if member is in sync with leader, and will error out otherwise
 	_, err := cli.MemberPromote(memPromoteCtx, member.ID)
 	if err == nil {
 		//Member successfully promoted
+		metrics.HasLearner.With(prometheus.Labels{}).Set(0)
```

Comment:
I see that `HasLearner` is set to `1` when `CheckIfLearnerPresent` is called which will return true when a learner is present in the cluster, but is set to `0` only when `DoPromoteMember` is called
This has a potential that a member might set its `HasLearner` metric to `1` and never set it back to `0` if it is already a voting member but finds another learner in the cluster